packages/plus/src/components/OfferList/components/Row.tsx

```diff
@@ -0,0 +1,344 @@
+'use client'
+
+import styled from '@emotion/styled'
+import { ArrowDownIcon, ArrowUpIcon } from '@ultraviolet/icons'
+import {
+  Badge as BadgeUV,
+  Button,
+  Checkbox,
+  List,
+  Radio,
+  Tooltip,
+} from '@ultraviolet/ui'
+import { Children, useCallback, useMemo } from 'react'
+import type { ComponentProps, ReactNode } from 'react'
+import { useOfferListContext } from '../OfferListProvider'
+import { Banner } from './Banner'
+
+const StyledBadge = styled(BadgeUV)`
+  position: absolute;
+  left: ${({ theme }) => theme.space[1]};
+  transform: translateY(-150%);
+  top: 0;
+  left: ${({ theme }) => theme.space[3]};
+`
+
+const BadgeContainer = styled.div`
+position: absolute;
+top: ${({ theme }) => theme.space[2]};
+`
+
+const NoPaddingCell = styled(List.Cell)`
+  padding: 0;
+
+  &:first-of-type {
+    padding-left: ${({ theme }) => theme.space['2']};
+  }
+
+`
+const StyledRow = styled(List.Row, {
+  shouldForwardProp: prop => !['selected', 'hasBanner'].includes(prop),
+})<{ selected: boolean; hasBanner: boolean }>`
+
+    ${({ theme, selected }) =>
+      selected
+        ? `td, td:first-child, td:last-child {
+      border-color: ${theme.colors.primary.border};
+    }`
+        : null}
+
+    &[aria-expanded='true'] {
+      ${({ theme, selected }) =>
+        selected
+          ? `td, td:first-child, td:last-child {
+      border-color: ${theme.colors.primary.border};
+    }`
+          : null}
+    }
+
+    ${({ hasBanner }) =>
+      hasBanner
+        ? `td, td:first-child {
+        border-bottom-left-radius: 0;
+        }
+        
+        td, td:last-child {
+          border-bottom-right-radius: 0;
+        }
+    `
+        : null}
+`
+
+const SelectableContainer = styled.div`
+  display: flex;
+`
+
+const CustomExpandable = styled('div', {
+  shouldForwardProp: prop => !['padding'].includes(prop),
+})<{ padding?: ComponentProps<typeof List.Row>['expandablePadding'] }>`
+    padding: ${({ theme, padding }) =>
+      padding ? theme.space[padding] : theme.space['2']};
+`
+
+type RowProps = ComponentProps<typeof List.Row> & {
+  banner?: {
+    text: ReactNode
+    sentiment?: 'neutral' | 'primary' | 'warning' | 'danger'
+  }
+  offerName: string
+  badge?: {
+    text: string
```

Comment:
I have a feeling that we will have icons and more complex stuff in those badges 😅